### PR TITLE
ETP-51 fix missing rsvp now

### PR DESCRIPTION
**Pull request by cliffordp**
Redo of https://github.com/moderntribe/event-tickets/pull/1523, which was based on https://github.com/moderntribe/event-tickets/tree/release/B20.01

New 1min video demo for _this_ PR: https://share.getcloudapp.com/geurRryA

[ETP-51]

[ETP-51]: https://moderntribe.atlassian.net/browse/ETP-51

### CODE REVIEW

File: src/Tribe/Assets.php [hunk 1]
```diff
@@ -18,7 +18,7 @@ public function enqueue_scripts() {
 				[ 'event-tickets-reset-css', 'reset.css' ],
 				[ 'event-tickets-tickets-css', 'tickets.css', [ 'dashicons', 'event-tickets-reset-css' ] ],
 				[ 'event-tickets-tickets-rsvp-css', 'rsvp.css', [] ],
-				[ 'event-tickets-tickets-rsvp-js', 'rsvp.js', [ 'jquery', 'jquery-ui-datepicker' ] ],
+				[ 'event-tickets-tickets-rsvp-js', 'rsvp.js', [ 'jquery' ] ],
```
Comment (tr1b0t) [inline src/Tribe/Assets.php]: * :no_entry_sign: Short array syntax is not allowed
* :no_entry_sign: Short array syntax is not allowed

<sub>hash: e7a50a8867246744676a1d24d73f85db</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 2]
```diff
@@ -20,34 +20,37 @@ class Tribe__Tickets__Attendee_Registration__Template extends Tribe__Templates {
 	public function hook() {
 
 		// Spoof the context
-		add_filter( 'the_posts', array( $this, 'setup_context' ), -10 );
+		add_filter( 'the_posts', [ $this, 'setup_context' ], -10 );
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: e5f8028389c369b12cab1af91a098a05</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 3]
```diff
@@ -20,34 +20,37 @@ class Tribe__Tickets__Attendee_Registration__Template extends Tribe__Templates {
...
 
 		// Set and remove the required body classes
-		add_action( 'wp', array( $this, 'set_body_classes' ) );
+		add_action( 'wp', [ $this, 'set_body_classes' ] );
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 2dcad43a9569c737fa27b726c68e0457</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 4]
```diff
@@ -20,34 +20,37 @@ class Tribe__Tickets__Attendee_Registration__Template extends Tribe__Templates {
...
 
 		/*
 		 * Choose the theme template to use. It has to have a higher priority than the
 		 * TEC filters (at 10) to ensure they do not usurp our rewrite here.
 		 */
-		add_filter( 'template_include', array( $this, 'set_page_template' ), 15 );
+		add_filter( 'template_include', [ $this, 'set_page_template' ], 15 );
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: c7cf9ef6db438e29575970169cb49301</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 5]
```diff
@@ -20,34 +20,37 @@ class Tribe__Tickets__Attendee_Registration__Template extends Tribe__Templates {
...
+
+		add_action( 'tribe_events_editor_assets_should_enqueue_frontend', [ $this, 'should_enqueue_frontend' ] );
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: f189d6803869805ea0ff9e6d580e610a</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 6]
```diff
@@ -20,34 +20,37 @@ class Tribe__Tickets__Attendee_Registration__Template extends Tribe__Templates {
...
+		add_action( 'tribe_events_views_v2_assets_should_enqueue_frontend', [ $this, 'should_enqueue_frontend' ] );
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 6d9916499229225bb6b2cf5c95b83dd9</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 7]
```diff
@@ -20,34 +20,37 @@ class Tribe__Tickets__Attendee_Registration__Template extends Tribe__Templates {
...
 
 		/*
 		 * Set the content of the page. Again, it has to have a higher priority than the
 		 * TEC filters (at 10) to ensure they do not usurp our rewrite here.
 		 */
-		add_action( 'loop_start', array( $this, 'set_page_content' ), 15 );
+		add_action( 'loop_start', [ $this, 'set_page_content' ], 15 );
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 3c4336ff71580baff189a69d790f4119</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 8]
```diff
@@ -20,34 +20,37 @@ class Tribe__Tickets__Attendee_Registration__Template extends Tribe__Templates {
...
 
 		// Modify the link for the edit post link
-		add_filter( 'edit_post_link', array( $this, 'set_edit_post_link' ) );
+		add_filter( 'edit_post_link', [ $this, 'set_edit_post_link' ] );
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: b13e02eb6ae62cbdce178735e1a55d91</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 9]
```diff
@@ -20,34 +20,37 @@ class Tribe__Tickets__Attendee_Registration__Template extends Tribe__Templates {
...
 
 		// Switcheroo for Genesis using the excerpt as we're saying we're on an archive
-		add_filter( 'genesis_pre_get_option_content_archive', array( $this, 'override_genesis_archive' ), 10, 2 );
+		add_filter( 'genesis_pre_get_option_content_archive', [ $this, 'override_genesis_archive' ], 10, 2 );
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: d65ad758330e5123a97243c658683cf1</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 10]
```diff
@@ -20,34 +20,37 @@ class Tribe__Tickets__Attendee_Registration__Template extends Tribe__Templates {
...
 		// Also keep content limit from truncating the form
-		add_filter( 'genesis_pre_get_option_content_archive_limit', array( $this, 'override_genesis_limit' ), 10, 2 );
+		add_filter( 'genesis_pre_get_option_content_archive_limit', [ $this, 'override_genesis_limit' ], 10, 2 );
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: dab44db9bc2f7052194fa20aa402725e</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 11]
```diff
@@ -20,34 +20,37 @@ class Tribe__Tickets__Attendee_Registration__Template extends Tribe__Templates {
...
 
 		// Modify the page title
-		add_filter( 'document_title_parts', array( $this, 'modify_page_title' ), 1000 );
-		add_filter( 'get_the_archive_title', array( $this, 'modify_archive_title' ), 1000 );
+		add_filter( 'document_title_parts', [ $this, 'modify_page_title' ], 1000 );
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 5b9c14f8651e2eb988cbeb28b39f88d1</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 12]
```diff
@@ -20,34 +20,37 @@ class Tribe__Tickets__Attendee_Registration__Template extends Tribe__Templates {
...
+		add_filter( 'get_the_archive_title', [ $this, 'modify_archive_title' ], 1000 );
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 830464c108f26efd7642b97712833957</sub>

File: src/Tribe/Attendee_Registration/Template.php [hunk 13]
```diff
@@ -152,6 +155,22 @@ public function set_page_template( $template ) {
 		return $template;
 	}
 
+	/**
+	 * Ensure we enqueue the frontend styles and scripts from our plugins on the AR page.
+	 *
+	 * @since TBD
+	 *
+	 * @param boolean $enqueue
```
Comment (tr1b0t) [inline src/Tribe/Attendee_Registration/Template.php]: :no_entry_sign: Missing parameter comment

<sub>hash: ff0a1579f6397989f45080a64852ab58</sub>

File: src/Tribe/Editor/Blocks/Tickets.php [hunk 14]
```diff
@@ -103,7 +103,7 @@ public function assets() {
 			$plugin,
 			'tribe-tickets-gutenberg-tickets',
 			'tickets-block.js',
-			[ 'jquery', 'jquery-ui-datepicker', 'wp-util-not-in-footer', 'wp-i18n' ],
+			[ 'jquery', 'wp-util-not-in-footer', 'wp-i18n' ],
```
Comment (tr1b0t) [inline src/Tribe/Editor/Blocks/Tickets.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 580fb7b9af81a0e18bdf28d276d496c1</sub>

File: src/Tribe/Events/Views/V2/Models/Tickets.php [hunk 15]
```diff
@@ -109,7 +110,16 @@ public function fetch_data() {
 			return $this->data;
 		}
 
-		if ( ! tribe_events_has_tickets_on_sale( $this->post_id ) ) {
+		$num_ticket_types_available = 0;
+		foreach( $this->all_tickets as $ticket ) {
+			if ( ! tribe_events_ticket_is_on_sale( $ticket ) ) {
+				continue;
+			}
```
Comment (tr1b0t) [inline src/Tribe/Events/Views/V2/Models/Tickets.php]: * :no_entry_sign: There must be a space between foreach and the first parenthesis '('
* :no_entry_sign: Space after opening control structure is required
* :no_entry_sign: No space before opening parenthesis is prohibited
* :no_entry_sign: Expected 1 space(s) after FOREACH keyword; 0 found

<sub>hash: e2db76660c0d00532277816ffebcb992</sub>

File: src/Tribe/Events/Views/V2/Service_Provider.php [hunk 16]
```diff
@@ -15,14 +15,9 @@ class Service_Provider extends \tad_DI52_ServiceProvider {
...
 
-		if ( ! tribe_events_tickets_views_v2_is_enabled() ) {
-			return;
-		}
-
-		$this->register_hooks();
+		add_action( 'tribe_events_bound_implementations', [ $this, 'register_hooks' ] );
 
 		// Register the SP on the container
 		$this->container->singleton( 'tickets.views.v2.provider', $this );
```
Comment (tr1b0t) [inline src/Tribe/Events/Views/V2/Service_Provider.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: fdee6a413a556f4c2ec6de4041820607</sub>

File: src/Tribe/Tickets_View.php [hunk 17]
```diff
@@ -41,6 +41,7 @@ public static function hook() {
 		add_action( 'parse_request', [ $myself, 'prevent_page_redirect' ] );
 		add_filter( 'the_content', [ $myself, 'intercept_content' ] );
 		add_action( 'parse_request', [ $myself, 'maybe_regenerate_rewrite_rules' ] );
+		add_filter( 'tribe_events_views_v2_bootstrap_should_display_single', [ $myself, 'intercept_views_v2_single_display' ], 15, 4 );
```
Comment (tr1b0t) [inline src/Tribe/Tickets_View.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 75769450186e0a1b868f21640265f1df</sub>

**Comment by cliffordp** [comment]
This PR is replaced by #1543

**Pull request closed without merging**